### PR TITLE
Load StorageManifest from KnownReplicasPayload

### PR DESCRIPTION
**Pull request by kenoir**
For https://github.com/wellcomecollection/platform/issues/4521

### CODE REVIEW

File: indexer/bag_indexer/src/main/resources/application.conf [hunk 1]
```diff
@@ -1,5 +1,8 @@
 aws.metrics.namespace=${?metrics_namespace}
 aws.sqs.queue.url=${?queue_url}
+aws.vhs.dynamo.tableName=${?vhs_table_name}
+aws.vhs.s3.bucketName=${?vhs_bucket_name}
+aws.vhs.s3.globalPrefix=${?vhs_bucket_name}
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/main/resources/application.conf:5]: The fact that you need these flags to do a read-only VHS is a bit of an antipattern. Not one to fix here, but one of the things that makes multiple apps accessing the VHS a pain.

File: indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagIndexerWorker.scala [hunk 2]
```diff
@@ -17,19 +20,39 @@ import scala.concurrent.Future
...
+    val bagId = BagId(
+      space = source.storageSpace,
+      externalIdentifier = source.externalIdentifier
+    )
+
+    val bagVersion = source.version
+
+    // TODO:
+    //  - actually handle errors
+    //  - fix either/future weirdness
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagIndexerWorker.scala:48]: Are you planning to fix these here, or defer them to the rearrangement of these apps?

Comment (kenoir) [reply indexer/bag_indexer/src/main/scala/uk/ac/wellcome/platform/archive/indexer/bags/BagIndexerWorker.scala:48]: oh here - this is a draft PR as of now

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala [hunk 3]
```diff
@@ -39,4 +40,62 @@ class BagIndexerFeatureTest
...
+    )
+
+    (payload, bagId.toString)
+  }
+
+  override def convertIndexedT(
+                                payload: KnownReplicasPayload
+                              ): IndexedStorageManifest = {
+    IndexedStorageManifest(storageManifest)
+  }
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala:74]: nit: you can drop the braces

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala [hunk 4]
```diff
@@ -39,4 +40,62 @@ class BagIndexerFeatureTest
...
+
+  override def withIndexerWorker[R](index: Index, queue: SQS.Queue)(
+    testWith: TestWith[
+      IndexerWorker[KnownReplicasPayload, StorageManifest, IndexedStorageManifest],
+      R
+    ]
+  )(implicit decoder: Decoder[KnownReplicasPayload]): R = {
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala:81]: nit: Do we actually need to pass this decoder? Don't we get it by importing `JsonUtil._`?

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala [hunk 5]
```diff
@@ -2,41 +2,206 @@ package uk.ac.wellcome.platform.archive.indexer.bag
...
+                               ): IndexedStorageManifest = {
+    IndexedStorageManifest(storageManifest)
+  }
+
+  override def withIndexerWorker[R](index: Index, queue: SQS.Queue)(
+    testWith: TestWith[
+      IndexerWorker[KnownReplicasPayload, StorageManifest, IndexedStorageManifest],
+      R
+    ]
+  )(implicit decoder: Decoder[KnownReplicasPayload]): R = {
```
Comment (alexwlchan) [inline indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala:87]: Ditto above. Could this be pushed into a shared fixture? Looks identical.

**Comment by alexwlchan** [comment]
How have you managed to conflict with master, given you're the only one working on it?

Are you being hunted by a rogue version of Kenny from a dystopian future?!

**Comment by kenoir** [comment]
> Are you being hunted by a rogue version of Kenny from a dystopian future?!

I always suspected this to be the case.

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala (hunk 4 above)
Comment (kenoir) [reply indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerFeatureTest.scala:82]: I think as this is overriding an existing function, we must define this.

File: indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala (hunk 5 above)
Comment (kenoir) [reply indexer/bag_indexer/src/test/scala/uk/ac/wellcome/platform/archive/indexer/bag/BagIndexerWorkerTest.scala:112]: You'd think. It's possible, but the inheritance here makes it awkward to compose sensibly. Probably needs a re-jig, will rethink in a future PR.

**Pull request merged**